src/test/java/org/broadinstitute/barclay/argparser/CommandLinePluginUnitTest.java

```diff
@@ -13,126 +8,188 @@
 import java.util.function.Predicate;
 
 /**
- * Test basic command line parser plugin functionality. Fully testing the plugin functionality requires
- * implementations of multiple classes and subclasses. The unit tests for GATKReadFilterPlugin
- * ReadFilterPluginUnitTest have much more extensive coverage of the plugin functionality since they
- * use a real (ReadFilter) plugin class hierarchy.
+ * Test command line parser plugin functionality.
  */
-public class CommandLineParserPluginUnitTest {
+public class CommandLinePluginUnitTest {
 
     public static class TestPluginBase {
     }
 
+    public static class TestPluginWithOptionalArg extends TestPluginBase {
+        public static final String optionalArgName = "optionalStringArg";
+
+        @Argument(fullName=optionalArgName, optional=true)
+        String optionalArg;
+    }
+
+    public static class TestPluginWithRequiredArg extends TestPluginBase {
+        public static final String requiredArgName = "requiredStringArg";
+
+        @Argument(fullName=requiredArgName, optional=false)
+        String requiredArg;
+    }
+
     public static class TestPlugin extends TestPluginBase {
         final static String argumentName = "argumentForTestPlugin";
         @Argument(fullName = argumentName, optional=true)
         Integer argumentForTestPlugin;
     }
 
-    public static class TestPluginDescriptor extends GATKCommandLinePluginDescriptor<TestPluginBase> {
+    public class TestPluginDescriptor extends CommandLinePluginDescriptor<TestPluginBase> {
```

Comment:
I'm not sure why that changed, but I had to restore it in the docgen branch anyway so I'll just restore it here too.